File: exercism/haskell/raindrops/src/Raindrops.hs
module Raindrops
  ( convert
  ) where

import Data.Maybe (fromMaybe)

convert :: Int -> String
convert n = fromMaybe (show n)
  $ mconcat [sound 3 "Pling", sound 5 "Plang", sound 7 "Plong"]
 where
  sound div noise | mod n div == 0 = Just noise
                  | otherwise      = Nothing

File: exercism/ocaml/raindrops/raindrops.ml
open Core

let sound n div str = match n mod div with 0 -> str | _ -> ""

let raindrop n =
  let sounds = sound n 3 "Pling" ^ sound n 5 "Plang" ^ sound n 7 "Plong" in
  match sounds with "" -> string_of_int n | _ -> sounds
